### PR TITLE
Helpers functions can now lookup by id or by keyname

### PR DESCRIPTION
**Pull request by tmulhern3**

### CODE REVIEW

**Comment by renier** [comment]
General thought: If you have the Id, you would normally just use _getObject_ on the particular service to get it. Rather than passing an object filter with the Id.

Are you looking to support when the conf user puts in Ids for these things?

**Comment by tmulhern3** [comment]
I could use the `getObject` functions however they don't currently exist. I think it would be overkill to make `getObject` functions for datacenters, routing types, routing methods, and health check types when we only need to retrieve the name/id pairing.

**Comment by tmulhern3** [comment]
As for the need for retrieving id based on name: I need to do a "reverse lookup" when performing a read (since the API call sends back id and the users tf code contains the name)

File: common/helpers.go [hunk 1]
```diff
@@ -41,15 +41,17 @@ func isInt(key interface{}) (bool, error) {
 }
 
 func getValueFromKey(client softlayer.Client, nameMask string, nameType string, nameTypeGet string, key interface{}, getById bool, lookupFunc lookup) (interface{}, error) {
-	var ObjectFilter string
+	var ObjectFilter, Url string
 	if getById {
+		Url = fmt.Sprintf("%s/%d/getObject.json", nameType, key.(int))
 		ObjectFilter = string(`{"id":{"operation":"` + strconv.Itoa(key.(int)) + `"}}`)
```
Comment (renier) [inline common/helpers.go:53]: ObjectFilter would not be needed in this case, I guess

**Pull request merged**